core/templates/dev/head/domain/collection/CollectionObjectFactory.js

```diff
@@ -203,6 +227,83 @@ oppia.factory('CollectionObjectFactory', [
       return skillList;
     };
 
+    // Gets a new ID for a skill. This should be of the same form as in the
+    // backend. This increments nextSkillId.
```

Comment:
Can you mention the specific file where this logic exists in the backend (should be one place)? Can you make sure that backend location also points here? Can you make sure we have targeted unit tests in both the backend and frontend to ensure the respective methods properly compute a new skill ID? We want to be really aggressive about making sure these two places of code are exactly correct, don't regress, and don't deviate from one another.